lib/matplotlib/tests/test_axes.py

```diff
@@ -2383,6 +2384,21 @@ def test_scatter_color(self):
         with pytest.raises(ValueError):
             plt.scatter([1, 2, 3], [1, 2, 3], color=[1, 2, 3])
 
+    @pytest.mark.parametrize('kwargs',
+                                [
+                                    {'cmap': 'gray'},
+                                    {'norm': mcolors.Normalize()},
+                                    {'vmin': 0},
+                                    {'vmax': 0}
+                                ])
+    def test_scatter_color_warning(self, kwargs):
+        # Warn for cases where 'cmap', 'norm', 'vmin', 'vmax'
+        # kwargs are being overridden
+        with pytest.warns():
+            plt.scatter([], [], **kwargs)
+        with warnings.catch_warnings():
+            plt.scatter([], [], c=[], **kwargs)
```

Comment:
I would expect this _not_ to warn as a `c` that is consistent with x and y was passed.